### PR TITLE
Using new Bluetooth scanning API to catch BT errors

### PR DESCRIPTION
**Pull request by stanleyguevara**

### CODE REVIEW

File: android-sdk/src/main/java/com/sensorberg/sdk/internal/AndroidBluetoothPlatform21.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    /**
+     * Returns a flag indicating whether Bluetooth is enabled.
+     *
+     * @return a flag indicating whether Bluetooth is enabled
+     */
+    @Override
+    public boolean isBluetoothLowEnergyDeviceTurnedOn() {
+        //noinspection SimplifiableIfStatement,SimplifiableIfStatement,SimplifiableIfStatement,SimplifiableIfStatement,SimplifiableIfStatement,SimplifiableIfStatement,SimplifiableIfStatement,SimplifiableIfStatement
+        return isBluetoothLowEnergySupported() && (bluetoothAdapter.isEnabled());
```
Comment (budius) [inline android-sdk/src/main/java/com/sensorberg/sdk/internal/AndroidBluetoothPlatform21.java]: I saw some weird stuff with `BluetoothAdapter.getDefaultAdapter()` returning null if u turn off BT on the device.
So on the safe side I would call everytime `getDefaultAdapter()` and null check.

File: android-sdk/src/main/java/com/sensorberg/sdk/internal/AndroidBluetoothPlatform21.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        }
+    }
+
+    @Override
+    public void stopLeScan() {
+        if (isBluetoothLowEnergySupported() && callback != null) {
+            try {
+                //noinspection deprecation old API compatability
+                scanner.stopScan(callback);
+            } catch (Exception sentBySysteminternally) {
```
Comment (budius) [inline android-sdk/src/main/java/com/sensorberg/sdk/internal/AndroidBluetoothPlatform21.java]: missing `callback = null`

File: android-sdk/src/main/java/com/sensorberg/sdk/scanner/AbstractScanner.java [hunk 3]
```diff
@@ -304,12 +335,26 @@ public void stop() {
         runLoop.sendMessage(ScannerEvent.SCAN_STOP_REQUESTED);
     }
 
-    @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
-    private class ScanCallback implements BluetoothAdapter.LeScanCallback {
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
```
Comment (budius) [inline android-sdk/src/main/java/com/sensorberg/sdk/scanner/AbstractScanner.java]: LOLLIPOP?

Comment (stanleyguevara) [inline android-sdk/src/main/java/com/sensorberg/sdk/scanner/AbstractScanner.java]: Aw, crap.

Comment (budius) [inline android-sdk/src/main/java/com/sensorberg/sdk/scanner/AbstractScanner.java]: Here we should create our own interface, internally `BluetoothPlatform` implements/contains the native listener and re-directs to our interface.

File: android-sdk/src/main/java/com/sensorberg/sdk/internal/AndroidBluetoothPlatform21.java (hunk 1 above)
Comment (stanleyguevara) [inline android-sdk/src/main/java/com/sensorberg/sdk/internal/AndroidBluetoothPlatform21.java]: This wasn't done before and it's calling the same API, should be ok.

**Pull request merged**